s3/proto/s3.proto

```diff
@@ -315,16 +315,53 @@ message BaseRequest {
 }
 
 message ListObjectsRequest {

```

Comment:
can we compose this out of smaller messages ?
like (only for example)
```
message LORSSEOpts{}
message LORVersionOpts{}

message LOROpts{
LORSSEOpts{}
LORVersionOpts{}

}
```

Reply:
Sorry, why need this?

Reply:
to make the struct and code modular...favor composition over one large struct